src/roi/Roi.ts

```diff
@@ -221,48 +221,48 @@ export class Roi {
       return 2 * Math.sqrt(this.surface / Math.PI);
     });
   }
-
+  /**
+   * Getter that calculates the number of holes in the ROI and their total surface
+   * It is used to calculate fillRatio
+   */
   get holesInfo() {
     return this.#getComputed('holesInfo', () => {
       return getHolesInfo(this);
     });
   }
 
-  getExternalIDs(): {
-    externalIDs: number[];
-    externalLengths: number[];
-  } {
+  getExternalIDs(): Border[] {
     // take all the borders and remove the internal one ...
-    let borders = this.borderIDs;
-    let lengths = this.borderLengths;
+    let borders = this.borders;
 
     this.#computed.externalLengths = [];
-    this.#computed.externalIDs = [];
+    this.#computed.externalBorders = [];
 
     let internals = this.internalIDs;
 
-    for (let i = 0; i < borders.length; i++) {
-      if (!internals.includes(borders[i])) {
-        this.#computed.externalIDs.push(borders[i]);
-        this.#computed.externalLengths.push(lengths[i]);
+    for (let border of borders) {
+      if (!internals.includes(border.connectedID)) {
+        const element: Border = {
+          connectedID: border.connectedID,
+          length: border.length,
+        };
+
+        this.#computed.externalBorders.push(element);
```

Comment:
We created the `getComputed` method so that we don't have to set properties on it ourselves.

Never set or get `#computed` except for in the `getComputed` method.